File: app/src/main/java/com/googlepractices/budgetly/ui/fragments/StatisticsFragment.java
package com.googlepractices.budgetly.ui.fragments;

import android.graphics.Color;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.fragment.app.Fragment;

import com.googlepractices.budgetly.R;
import com.googlepractices.budgetly.databinding.FragmentStatisticsBinding;
import org.eazegraph.lib.charts.PieChart;
import org.eazegraph.lib.models.PieModel;

class StatisticsFragment extends Fragment {

    TextView tvR, tvPython, tvCPP, tvJava;
    PieChart pieChart;
    public StatisticsFragment() {
        // Required empty public constructor
    }

    public static StatisticsFragment newInstance(String param1, String param2) {
        StatisticsFragment fragment = new StatisticsFragment();
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Link those objects with their
        // respective id's that
        // we have given in .XML file

        pieChart = getActivity().findViewById(R.id.pie_chart);

        // Creating a method setData()
        // to set the text in text view and pie chart
        setData();
    }

    private void setData()
    {
        // Set the data and color to the pie chart
        pieChart.addPieSlice(
                new PieModel(
                        "R",
                        Integer.parseInt(tvR.getText().toString()),
                        Color.parseColor("#FFA726")));
        pieChart.addPieSlice(
                new PieModel(
                        "Python",
                        Integer.parseInt(tvPython.getText().toString()),
                        Color.parseColor("#66BB6A")));
        pieChart.addPieSlice(
                new PieModel(
                        "C++",
                        Integer.parseInt(tvCPP.getText().toString()),
                        Color.parseColor("#EF5350")));
        pieChart.addPieSlice(
                new PieModel(
                        "Java",
                        Integer.parseInt(tvJava.getText().toString()),
                        Color.parseColor("#29B6F6")));

        // To animate the pie chart
        pieChart.startAnimation();
    }



}

File: app/src/main/java/com/googlepractices/budgetly/StatisticsFragment.kt
package com.googlepractices.budgetly

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.googlepractices.budgetly.databinding.FragmentExpensesBinding
import com.googlepractices.budgetly.databinding.FragmentStatisticsBinding

class StatisticsFragment: Fragment() {

    private var _binding: FragmentStatisticsBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        _binding = FragmentStatisticsBinding.inflate(inflater, container, false)
        return binding.root
    }

}